### PR TITLE
Add support for tap alert button by name

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/8361

### CODE REVIEW

File: lib/commands/alert.js [hunk 1]
```diff
@@ -39,10 +39,13 @@ commands.setAlertText = async function (text) {
...
 };
 
-commands.postAcceptAlert = async function () {
+commands.postAcceptAlert = async function (opts = {}) {
   try {
     let method = 'POST';
-    let endpoint = `/alert/accept`;
+    let endpoint = '/alert/accept';
+    if (opts.buttonLabel) {
+      endpoint += `?name=${encodeURIComponent(opts.buttonLabel)}`;
```
Comment (imurchie) [inline lib/commands/alert.js]: Why add it to the query instead of adding params to the proxy call?

Comment (mykola-mokhnach) [inline lib/commands/alert.js]: how can I add them as query parameters - is there any examples? I though proxyCommand only accepts POST form parameters.

Comment (imurchie) [inline lib/commands/alert.js]: It does. WDA requires query parameters?

Comment (mykola-mokhnach) [inline lib/commands/alert.js]: yes:

```objc
+ (id<FBResponsePayload>)handleAlertDismissCommand:(FBRouteRequest *)request
{
  FBSession *session = request.session;
  NSString *name = request.parameters[@"name"];
  FBAlert *alert = [FBAlert alertWithApplication:session.application];
  NSError *error;
    
  if (!alert.isPresent) {
    return FBResponseWithStatus(FBCommandStatusNoAlertPresent, nil);
  }
  if ([name length] != 0) {
    if (![alert clickAlertButton:name error:&error]) {
      return FBResponseWithError(error);
    }
  } else if (![alert dismissWithError:&error]) {
    return FBResponseWithError(error);
  }
  return FBResponseWithOK();
}

```

Comment (imurchie) [inline lib/commands/alert.js]: Is that a mistake? It seems incommensurate with the rest of their API.

Comment (mykola-mokhnach) [inline lib/commands/alert.js]: At least this is how it works right now %)

Comment (imurchie) [inline lib/commands/alert.js]: Weird. But ok. If another method crops up like that, we should probably refactor `proxyCommand` to handle it.

Comment (mykola-mokhnach) [inline lib/commands/alert.js]: we could create a PR in WDA to change name into argument instead of paramter

Comment (imurchie) [inline lib/commands/alert.js]: Yeah. It would probably have to handle both cases in case people are actually using it.

Want to do that?

Comment (jlipps) [inline lib/commands/alert.js]: yes that is very much outside the webdriver spec to have query params. you either have params in the url or in the json post payload.

File: test/unit/commands/alert-specs.js [hunk 2]
```diff
@@ -43,4 +43,35 @@ describe('alert commands', () => {
...
+    const commandName = 'alert';
+
+    it('should reject request to WDA if action parameter is not supported', async () => {
+      await driver.execute(`mobile: ${commandName}`, {action: 'blabla'})
+        .should.be.rejectedWith(/should be either/);
+    });
+
+    it('should send accept alert request to WDA with encoded button label', async () => {
+      const buttonLabel = 'some label';
+      await driver.execute(`mobile: ${commandName}`, {action: 'accept', buttonLabel: buttonLabel});
```
Comment (jlipps) [inline test/unit/commands/alert-specs.js]: `{action: 'accept', buttonLabel}`

**Comment by mykola-mokhnach** [comment]
Create WDA PR https://github.com/facebook/WebDriverAgent/pull/566

File: test/unit/commands/alert-specs.js (hunk 2 above)
Comment (imurchie) [inline test/unit/commands/alert-specs.js]: This is what is causing Travis to fail.

Comment (mykola-mokhnach) [inline test/unit/commands/alert-specs.js]: interesting - why I don't see any errors on my local system?

Comment (imurchie) [inline test/unit/commands/alert-specs.js]: Did you run `gulp eslint`?

Comment (mykola-mokhnach) [inline test/unit/commands/alert-specs.js]: Ok. Now I got it. The next commit fixed that

**Comment by mykola-mokhnach** [comment]
@imurchie 
The WDA PR has been recently merged (fast like never o_O)
We could incorporate it and merge this PR after.

**Comment by jlipps** [comment]
Yeah, let's just update things here since we can, before merging.

**Comment by imurchie** [comment]
Upstream change is in.

**Comment by mykola-mokhnach** [comment]
Can you please merge it then @imurchie

**Comment by imurchie** [comment]
Published in `appium-xcuitest-driver@2.28.6`.

**Pull request merged**